src/components/app.js

```diff
@@ -19,51 +20,88 @@ import NewIssue from 'components/issues/new-issue/NewIssue';
 
 import {
   getUserInfoWithToken,
-  saveTokenToLocalStorage
+  saveTokenToLocalStorage,
+  getUserTokenWithCode
 } from 'actions/user.actions';
 
 type ConnectedProps = {
   getUserInfoWithToken: () => void,
-  saveTokenToLocalStorage: () => void
+  saveTokenToLocalStorage: () => void,
+  isAuthenticated: string | null
 };
 
 type OwnProps = {};
 
 class App extends React.Component<ConnectedProps & OwnProps> {
   componentWillMount() {
     const user = get('user', loadFromStorage('auth'));
-    if (user) {
+
+    if (user && !isNull(user.token)) {
       this.props.saveTokenToLocalStorage(user);
       this.props.getUserInfoWithToken(user);
+    } else {
+      const searchParams = get('search', this.props.location);
```

Comment:
This is code that should be moved to the separate route we talked about which handles getting the token from github so this behaviour will be in 1 place and won't affect the entire application